### ISSUE TITLE
With RTL, the RichText content moves but the widget does not

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#62550]
See repro:

```dart
Container(
    padding: EdgeInsets.all(matGrid * 2),
    child: Column(
        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text('LTR Text Span (Default TextWidthBasis):',
              style: Theme.of(context).textTheme.caption),
          Container(
              decoration: BoxDecoration(border: Border.all(width: 1)),
              child: Text.rich(
                TextSpan(children: [
                  TextSpan(text: 'this is a line of text with words'),
                ]),
              )),
          Container(
              decoration: BoxDecoration(border: Border.all(width: 1)),
              child: Text.rich(
                TextSpan(children: [
                  TextSpan(
                      text:
                          'this is a line of text with words followed by areallylongwordthatneedstobewrapped'),
                ]),
              )),
          Text('RTL Text Span (Default TextWidthBasis):',
              style: Theme.of(context).textTheme.caption),
          Container(
              decoration: BoxDecoration(border: Border.all(width: 1)),
              child: Text.rich(
                  TextSpan(children: [
                    TextSpan(text: 'this is a line of text with words'),
                  ]),
                  textDirection: TextDirection.rtl)),
          Container(
              decoration: BoxDecoration(border: Border.all(width: 1)),
              child: Text.rich(
                  TextSpan(children: [
                    TextSpan(
                        text:
                            'this is a line of text with words followed by areallylongwordthatneedstobewrapped'),
                  ]),
                  textDirection: TextDirection.rtl)),
          Text('LTR Text Span (TextWidthBasis.longestLine):',
              style: Theme.of(context).textTheme.caption),
          Container(
              decoration: BoxDecoration(border: Border.all(width: 1)),
              child: Text.rich(
                  TextSpan(children: [
                    TextSpan(text: 'this is a line of text with words'),
                  ]),
                  textWidthBasis: TextWidthBasis.longestLine)),
          Container(
              decoration: BoxDecoration(border: Border.all(width: 1)),
              child: Text.rich(
                  TextSpan(children: [
                    TextSpan(
                        text:
                            'this is a line of text with words followed by areallylongwordthatneedstobewrapped'),
                  ]),
                  textWidthBasis: TextWidthBasis.longestLine)),
          Text('RTL Text Span (TextWidthBasis.longestLine):',
              style: Theme.of(context).textTheme.caption),
          Container(
              decoration: BoxDecoration(border: Border.all(width: 1)),
              child: Text.rich(
                  TextSpan(children: [
                    TextSpan(text: 'this is a line of text with words'),
                  ]),
                  textWidthBasis: TextWidthBasis.longestLine,
                  textDirection: TextDirection.rtl)),
          Container(
              decoration: BoxDecoration(border: Border.all(width: 1)),
              child: Text.rich(
                  TextSpan(children: [
                    TextSpan(
                        text:
                            'this is a line of text with words followed by areallylongwordthatneedstobewrapped '),
                  ]),
                  textWidthBasis: TextWidthBasis.longestLine,
                  textDirection: TextDirection.rtl)),
        ]));
```

Produces the following screen (screen does not map to code directly. Customer reported two issues in a single repro and I edited the code):

![image](https://user-images.githubusercontent.com/6687929/88883445-37aea900-d1e9-11ea-9f97-b1b97c723261.png)

**Comment by HansMuller** [comment]
CC @chunhtai

**Comment by mehmetf** [comment]
Update from customer:

I found that the RTL issue appears to happen even with regular Text widgets, and I believe the issue is with the positioning of the text, not the widget, which perhaps makes sense if it's happening for both Text and RichText widgets. I will see if I can make another code example when I get a chance.

A workaround:

RTL Text - I created an invisible text widget with TextDirection.ltr and TextWidthBasis.longestLine and added it to a Container with the BoxDecoration I want to surround the RTL text with. Then I created the RTL text widget with TextWidthBasis.parent and layer them in a Stack. I will probably optimize this by using TextPainter, as you mentioned, to size the Container in lieu of the invisible text widget.
